archethic/contracts/factory.exs

```diff
@@ -151,43 +176,40 @@ export fun get_chargeable_htlc(end_time, user_address, pool_address, secret_hash
     actions triggered_by: transaction, on: reveal_secret(secret) do
       Contract.set_type "transfer"
       #{valid_transfer_code}
-      Contract.set_code ""
-    end
 
-    export fun get_provision_data() do
+      Contract.set_code \\\"""
+      @version 1
+
+      export fun info() do
         [
-          evm_contract: "\#{evm_contract}",
-          signature: [
-            r: 0x\#{signature.r},
-            s: 0x\#{signature.s},
-            v: \#{signature.v}
-          ]
+          evm_contract: \#{evm_contract},
+          evm_pool: \#{evm_pool},
+          ae_pool: 0x#{pool_address},
+          status: 1 # WITHDRAWN
         ]
       end
-  """
-
-  """
-  @version 1
+      \\\"""
+    end
 
-  condition triggered_by: transaction, on: provision(_evm_contract, _url, _signature), as: [
-		previous_public_key: (
-	    # Transaction is not yet validated so we need to use previous address
-		  # to get the genesis address
-		  previous_address = Chain.get_previous_address()
-		  Chain.get_genesis_address(previous_address) == 0x#{pool_address}
-	  )
-  ]
+    export fun get_provision_data() do
```

Comment:
If you remove from this method the evm contract, perhaps it's better to call the method like before:  get_provision_signature ?